meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -115,6 +118,86 @@ public void create(String key, Object data, MetaClientInterface.EntryMode mode)
     }
   }
 
+  @Override
+  public void recursiveCreate(String key, T data, EntryMode mode) {
+    iterativeCreateHelperTwo(key, data, mode, -1);
+  }
+
+  @Override
+  public void recursiveCreateWithTTL(String key, T data, long ttl) {
+    iterativeCreateHelperTwo(key, data, EntryMode.TTL, ttl);
+  }
+
+  private void iterativeCreateHelper(String key, T data, EntryMode mode, long ttl) {
```

Comment:
Why we need this? It is not used anywhere.

Reply:
Mentioned in comment above, It was an alternative to see which one you preferred. I've removed it.